src/seahawk/seahawk_deck/set_remote_params.py

```diff
@@ -0,0 +1,85 @@
+
+Abstracts requesting to set another node's parameters via service to class
+'SetRemoteParams'. To import this class to a program on the deck use 
+'from .set_remote_params import SetRemoteParams'
+
+Copyright (C) 2022-2023 Cabrillo Robotics Club
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+Cabrillo Robotics Club
+6500 Soquel Drive Aptos, CA 95003
+cabrillorobotics@gmail.com
+"""
+# ROS client library
+from rclpy.node import Node 
+
+# Type hints
+from typing import Any
+
+# Parameters
+from rcl_interfaces.srv import SetParameters
+from rclpy.parameter import Parameter
+
+class RemoteParamsClient():
+    """
+    Abstracts requesting to set another node's parameters via service to a class. 
+    """
+    def __init__(self, this_node: Node, other_node_name: str):
+        """
+        Set up request to set another node's parameters via service named '/other_node_name/set_parameters'
+    
+        Args:
+            this_node: Node which makes the request to set another node's parameters. Client node
+            other_node_name: Name of the node whose parameters should be set. Service node
+        """
+        # Create service name. Services to set parameters must be named as '/other_node_name/set_parameters'
+        srv_name = "/" + other_node_name + "/set_parameters"
+
+        # Set up client to remotely set parameters of another node using a service
+        self.__cli = this_node.create_client(SetParameters, srv_name)
+
+        # Try to connect to the service (wait one second between attempts)
+        while not self.__cli.wait_for_service(timeout_sec=1.0):
+            this_node.get_logger().info(f"{srv_name} service not available, waiting again...")
+        
+        # Create a set parameter request object
+        self.__req = SetParameters.Request()
+
+        # List to store parameters before sending them
+        self.__param_list = []
+
+    def update_params(self, param_name: str, param_value: Any):
+        """
+        Appends a single new parameter given its name and new value to the list of parameters to set
+
+        Args: 
+            param_name: Name of the parameter
+            param_value: Value of the parameter (note must of a be a valid parameter type)
+        """
+        self.__param_list.append(Parameter(name=param_name, value=param_value).to_parameter_msg())
+	
+    def send_params(self) -> (Any | None):
```

Comment:
This is great code Steph. It makes it way easier to grab and hold onto a remote parameter that you're likey to change often. Here's something to consider. It looks like users of this have to:

```python 
p.update_params('foo', 'bar')
p.update_params('thing', 'bak')
p.send_params()
```

If you liked, you could combine the calls by making `send_params` take a dictionary:

```python
p.send_params({
  'foo': 'bar', 
  'thing': 'bak',
})
```

Just something to consider.

Reply:
This history is going to be rewritten as we discussed a few minutes ago, so this review will be lost. Sorry! Normally I wouldn't ever rewrite history on an active PR but this is emergency mode I think.

Reply:
Thanks!! In the current use cases I can think of, we are only going to be updating one parameter at a time, what would be an instance where we would want to update multiple parameters? I could not think of a case when writing the code, but knew maybe it could be a possibility and thus allowed the user to call `update_params()` multiple times. In my opinion it is not necessary to optimize for an unlikely case but let me know if there is something I am not considering

Reply:
The main issue with parameters is there are only a few fixed types. If you wanted to use the thrust matrix as a parameter, for example, it would have to be treated as an array with separate parameter(s) to describe how many columns/rows it has.

Reply:
Can't help but give a list comprehension for dealing with that:

`self.__req.parameters = [Parameter(name=param_name, value=param_value).to_parameter_msg() for param_name, param_value in self.__param_list.items()]`

Reply:
That isn't an issue @OrionOth, at least in regards to the motor configuration matrix. The parameter that handles variable center of mass is a three element array that indicates the offset of the center of mass in (x, y, z).